src/vscode-dts/vscode.proposed.envCollectionWorkspace.d.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/171173
+
+	export interface EnvironmentVariableMutator {
+		readonly type: EnvironmentVariableMutatorType;
+		readonly value: string;
+		readonly scope: EnvironmentVariableScope | undefined;
```

Comment:
```suggestion
		readonly scope?: EnvironmentVariableScope;
```

Was there a reason we went with this instead of the above?

Reply:
I've made it optional for internal use, keeping it mandatory for external API.